framework/include/ble_manager/ble_errno.h

```diff
@@ -0,0 +1,195 @@
+
+/* A parameter value requested is outside the mandatory range of parameters for the given HCI command or LMP PDU. */
+#define BLE_ERR_PARAM_OUT_OF_RANGE              0x30  /* 48 */
+
+#define BLE_ERR_ROLE_SWITCH_PANDING             0x32  /* 50 */
+#define BLE_ERR_RESERVED_SLOT_VIOLATION         0x34  /* 52 */
+#define BLE_ERR_ROLE_SWITCH_FAILED              0x35  /* 53 */
+#define BLE_ERR_EXT_INQUIRY_RSP_TOO_LARGE       0x36  /* 54 */
+
+/* The IO capabilities request or response was rejected because the sending Host does not support Secure Simple Pairing even though the receiving Link Manager does. */
+#define BLE_ERR_SSP_NOT_SUPPORTED_BY_HOST       0x37  /* 55 */
+
+/* The Host is busy with another pairing operation and unable to support the requested pairing. The receiving device should retry pairing again later. */
+#define BLE_ERR_HOST_BUSY_PAIRING               0x38  /* 56 */
+
+/* The Controller could not calculate an appropriate value for the Channel selection operation. */
+#define BLE_ERR_REJECT_NO_SUITABLE_CHANN        0x39  /* 57 */
+
+/* Operation was rejected because the controller is busy and unable to process the request. */
+#define BLE_ERR_CONTROLLER_BUSY                 0x3A  /* 58 */
+
+/* Remote device terminated the connection because of an unacceptable connection interval. */
+#define BLE_ERR_UNACCEPTABLE_CONN_INTERVAL      0x3B  /* 59 */
+
+/* Advertising for a fixed duration completed or, for directed advertising, that advertising completed without a connection being created. */
+#define BLE_ERR_DIRECTED_ADV_TIMEOUT            0x3C  /* 60 */
+
+/* Connection was terminated because the Message Integrity Check (MIC) failed on a received packet. */
+#define BLE_ERR_MIC_FAILURE                     0x3D  /* 61 */
+
+/* LL initiated a connection but the connection has failed to be established. Controller did not receive any packets from remote end. */
+#define BLE_ERR_FAIL_TO_ESTABLISH_CONN          0x3E  /* 62 */
+
+/* The MAC of the 802.11 AMP was requested to connect to a peer, but the connection failed. */
+#define BLE_ERR_MAC_CONN_FAIL                   0x3F  /* 63 */
+
+/* The master, at this time, is unable to make a coarse adjustment to the piconet clock, using the supplied parameters. Instead the master will attempt to move the clock using clock dragging. */
+#define BLE_ERR_COARSE_CLOCK_ADJUST_REJECTED    0x40
+
+#define BLE_ERR_TYPE0_SUBMAP_NOT_DEFINED        0x41
+
+/* A command was sent from the Host that should identify an Advertising or Sync handle, but the Advertising or Sync handle does not exist. */
+#define BLE_ERR_UNKNOWN_ADVERTISING_IDENTIFIER  0x42
+
+/* Number of operations requested has been reached and has indicated the completion of the activity (e.g., advertising or scanning). */
+#define BLE_ERR_LIMIT_REACHED                   0x43
+
+/* A request to the Controller issued by the Host and still pending was successfully canceled. */
+#define BLE_ERR_OPERATION_CANCELLED_BY_HOST     0x44
```

Comment:
1. Too many type of error, i think user doesn't need most of these....(even android ble doesn't inform error reason to user as I know)
2. If you brought these constant from driver, you should follow below to prevent error when driver modified.
```
os/include/ble.h
#define ERROR_LIMIT_REACHED 0x43
...

framework/include/ble_manager/ble_errno.h
#include "ble.h"
#define BLE_ERR_TYPE_LIMIT_REACHED     ERROR_LIMIT_REACHED
```